react/OrderItems.tsx

```diff
@@ -48,53 +48,70 @@ export const OrderItemsProvider = compose(
     renderComponent(EmptyState)
   )
 )(({ children, ItemListQuery, UpdateItem }: OrderItemsProviderProps) => {
-  const { enqueue } = useOrderManager()
+  const { enqueue, listen } = useOrderManager()
 
   const [itemList, setItemList] = useState(ItemListQuery.cart.items)
 
-  const updateItem = (index: number, quantity: number) => {
-    lastUpdateToken.outdated = true
-    const token = (lastUpdateToken = { outdated: false })
-
-    const updatedList =
-      quantity === 0
-        ? [...itemList.slice(0, index), ...itemList.slice(index + 1)]
-        : adjust(index, item => ({ ...item, quantity }), itemList)
-
-    setItemList(updatedList)
-
-    const task = async () => {
-      const {
-        data: {
-          updateItems: { items },
-        },
-      } = await UpdateItem({
-        variables: {
-          orderItems: [
-            {
-              index,
-              quantity,
-            },
-          ],
-        },
-      })
-
-      if (!token.outdated) {
-        setItemList(items)
+  const isQueueBusy = useRef(false)
+  useEffect(() => {
+    const unlisten = listen('Pending', () => (isQueueBusy.current = true))
+    return unlisten
+  })
+  useEffect(() => {
+    const unlisten = listen('Fulfilled', () => (isQueueBusy.current = false))
+    return unlisten
+  })
+
+  const updateItem = useMemo(
```

Comment:
You can use useCallback instead

It's just like useMemo, but to create memoized functions

Reply:
TIL how to react hooks